lib/api/CAnomalyJob.cc

```diff
@@ -1060,6 +1060,15 @@ bool CAnomalyJob::backgroundPersistState(CBackgroundPersister& backgroundPersist
     return true;
 }
 
+bool CAnomalyJob::runForegroundPersist(core::CDataAdder& persister) {
+    LOG_INFO(<< "Foreground persist commencing...");
+
+    // Finalise the processor so it gets a chance to write any remaining results
+    this->finalise();
```

Comment:
I'm not sure it's a good idea to call `finalise` more than once.  It might confuse a future maintainer as there are other `finalise` methods in the the code that really must only be called once.

But also this is changing the behaviour for persisting quantiles.  Previously quantiles were persisted:

1. On a significant change to the highest ever score
2. Periodically on the quantile persistence interval
3. On graceful close

But by calling `finalise` here you're changing that to:

1. On a significant change to the highest ever score
2. Periodically on the quantile persistence interval
3. On graceful close
4. Periodically on the state persistence interval

That leads to the question of why we'd have separate persistence intervals for quantiles and state.

To be more consistent with background persist it's probably better to call `pruneAllModels` here instead of `finalise`.